tasks/main.yml

```diff
@@ -67,12 +67,14 @@
   register: firewall_lib_result
 
 - name: gather firewalld configuration
-  when: not firewall | bool
+  when: not firewall | bool and not __firewall_previous_replaced | bool
```

Comment:
Why do you need to check for `__firewall_previous_replaced`?

Reply:
Because it collects the fact if you specify only previous replaced, the prior loop strips `previous` from firewall from what I've noticed when writing the test

Reply:
hmm - if that loop is modifying the `firewall` variable in place, that's really bad - can you put a
```yaml
- name: show firewall
  debug:
    msg: firewall is {{ firewall | to_nice_json }}
```
just after the call to `firewall_lib`?

Reply:
Looks like I was incorrect, firewalld is not altered like i thought it was. My mistake

Reply:
Yeah, blocked the two ansible fact plays into one, then placed the conditional on the block, and that fixed the problem I was having, pushing changes up now after removing the debug plays.